From 26fbdaacf367799ea1508a596ffaca8233687e1a Mon Sep 17 00:00:00 2001
From: Hanna Hawa <hannah@marvell.com>
Date: Thu, 20 Aug 2015 19:31:48 +0300
Subject: [PATCH 0357/1240] mvebu: kconfig: added Kconfig menu for all mvebu
 boards

- Added config for development and customer boards
- Added config for all MVEBU boards

Change-Id: I3d0b9b6b4038f5befe1a6a50515d8a99ab22d6dc
Signed-off-by: Hanna Hawa <hannah@marvell.com>
Reviewed-on: http://vgitil04.il.marvell.com:8080/23006
Tested-by: Star_Automation <star@marvell.com>
---
 arch/arm/Kconfig                           |  2 +-
 board/mvebu/Kconfig                        | 26 +++++++++++++++++++++++
 board/mvebu/armada38x/Kconfig              | 34 ++++++++++++++++++++++++++++++
 board/mvebu/armada38x/Makefile             |  6 +++---
 board/mvebu/armada38x/armada38x.c          | 10 +++++++++
 board/mvebu/armada8k/Kconfig               | 24 +++++++++++++++++++++
 board/mvebu/armada8k/Makefile              |  4 ++--
 board/mvebu/armada8k/armada8021-pxp.c      |  8 +++++++
 board/mvebu/armadalp/Kconfig               | 34 ++++++++++++++++++++++++++++++
 board/mvebu/armadalp/Makefile              |  6 +++---
 board/mvebu/armadalp/armadalp.c            | 10 +++++++++
 board/mvebu/common/Kconfig                 | 32 ----------------------------
 board/mvebu/common/Makefile                |  1 -
 configs/mvebu_apn806_defconfig             |  2 +-
 configs/mvebu_apn806_pd_defconfig          |  3 +--
 configs/mvebu_armada38x_defconfig          |  2 +-
 configs/mvebu_armada7040_pd_defconfig      |  1 -
 configs/mvebu_armada8040_pd_defconfig      |  1 -
 configs/mvebu_armadalp_defconfig           |  5 +++--
 configs/mvebu_armadalp_palladium_defconfig |  4 ++--
 20 files changed, 163 insertions(+), 52 deletions(-)
 create mode 100644 board/mvebu/Kconfig
 create mode 100644 board/mvebu/armada38x/Kconfig
 create mode 100644 board/mvebu/armada8k/Kconfig
 create mode 100644 board/mvebu/armadalp/Kconfig
 delete mode 100644 board/mvebu/common/Kconfig

diff --git a/arch/arm/Kconfig b/arch/arm/Kconfig
index 422c46d..6cc5001 100644
--- a/arch/arm/Kconfig
+++ b/arch/arm/Kconfig
@@ -852,7 +852,7 @@ source "arch/arm/cpu/armv7/zynq/Kconfig"
 source "arch/arm/cpu/armv7/Kconfig"
 
 source "arch/arm/cpu/mvebu-common/Kconfig"
-source "board/mvebu/common/Kconfig"
+source "board/mvebu/Kconfig"
 
 source "board/aristainetos/Kconfig"
 source "board/BuR/kwb/Kconfig"
diff --git a/board/mvebu/Kconfig b/board/mvebu/Kconfig
new file mode 100644
index 0000000..99628a7
--- /dev/null
+++ b/board/mvebu/Kconfig
@@ -0,0 +1,26 @@
+menu "MVEBU Board Support"
+
+config DEVEL_BOARD
+	bool "Support MVEBU development boards"
+	depends on MVEBU
+	default n
+	---help---
+	  Say Y here to add support for Marvell MVEBU
+	  development boards.
+	  You need this only if you work on a Marvell board.
+	  If not, keep this off to reduce code size
+
+config CUSTOMER_BOARD
+	bool "Support MVEBU customer boards"
+	depends on MVEBU
+	default n
+	help
+	  Say Y here to add support for customer boards.
+	  You need this only if you work on a spcific
+	  customer board.
+	  If not, keep this off to reduce code size
+
+source "board/mvebu/armada8k/Kconfig"
+source "board/mvebu/armada38x/Kconfig"
+source "board/mvebu/armadalp/Kconfig"
+endmenu
diff --git a/board/mvebu/armada38x/Kconfig b/board/mvebu/armada38x/Kconfig
new file mode 100644
index 0000000..6250c44
--- /dev/null
+++ b/board/mvebu/armada38x/Kconfig
@@ -0,0 +1,34 @@
+menu "Armada 38x boards"
+	depends on TARGET_ARMADA_38X
+
+config ARMADA_38X_DB
+	bool "Support Marvell Armada-38x DB board"
+	depends on DEVEL_BOARD
+	default y
+	help
+	  Choose this option to add
+	  support for Marvell Armada-38x
+	  DB board. Say N unless you really
+	  work on this board
+
+config ARMADA_38X_RD
+	bool "Support Marvell Armada-38x RD board"
+	depends on DEVEL_BOARD
+	default y
+	help
+	  Choose this option to add
+	  support for Marvell Armada-38x
+	  RD board. Say N unless you really
+	  work on this board
+
+config ARMADA_38X_CUSTOMER
+	bool "Support Marvell Armada-38x Customer board"
+	depends on CUSTOMER_BOARD
+	default y
+	help
+	  Choose this option to add
+	  support for Marvell Armada-38x
+	  Customer board. Say N unless you really
+	  work on this board
+
+endmenu
diff --git a/board/mvebu/armada38x/Makefile b/board/mvebu/armada38x/Makefile
index c0afbe0..8acecdb 100644
--- a/board/mvebu/armada38x/Makefile
+++ b/board/mvebu/armada38x/Makefile
@@ -17,6 +17,6 @@
 #
 
 obj-y	+= armada38x.o
-obj-y	+= db-88f6820-info.o
-obj-y	+= rd-88f6820-info.o
-obj-y	+= customer-board.o
+obj-$(CONFIG_ARMADA_38X_DB)	+= db-88f6820-info.o
+obj-$(CONFIG_ARMADA_38X_RD)	+= rd-88f6820-info.o
+obj-$(CONFIG_ARMADA_38X_CUSTOMER) += customer-board.o
diff --git a/board/mvebu/armada38x/armada38x.c b/board/mvebu/armada38x/armada38x.c
index 00a3b46..5f5a16d 100644
--- a/board/mvebu/armada38x/armada38x.c
+++ b/board/mvebu/armada38x/armada38x.c
@@ -28,14 +28,24 @@
 struct mvebu_board_info *mvebu_board_info_get(enum fdt_compat_id compat_id)
 {
 	switch (compat_id) {
+#ifdef CONFIG_ARMADA_38X_DB
 	case COMPAT_MVEBU_ARMADA_38X_DB:
 		return &a38x_db_info;
+#endif
+#ifdef CONFIG_ARMADA_38X_RD
 	case COMPAT_MVEBU_ARMADA_38X_RD:
 		return &a38x_rd_info;
+#endif
+#ifdef CONFIG_ARMADA_38X_CUSTOMER
 	case COMPAT_MVEBU_ARMADA_38X_CUSTOMER:
 		return &a38x_customer_info;
+#endif
 	default:
 		error("Missing board information for compatible string = %d\n", compat_id);
+#ifdef CONFIG_ARMADA_38X_DB
 		return &a38x_db_info;
+#else
+		return NULL;
+#endif
 	}
 }
diff --git a/board/mvebu/armada8k/Kconfig b/board/mvebu/armada8k/Kconfig
new file mode 100644
index 0000000..dd5e790
--- /dev/null
+++ b/board/mvebu/armada8k/Kconfig
@@ -0,0 +1,24 @@
+menu "Armada 8k boards"
+	depends on TARGET_ARMADA_8K
+
+config APN_806_DB
+	bool "Support Marvell Armada-8K DB board"
+	depends on DEVEL_BOARD
+	default y
+	help
+	  Choose this option to add
+	  support for Marvell Armada-8K
+	  DB board. Say N unless you really
+	  work on this board
+
+config APN_806_RD
+	bool "Support Marvell Armada-8K RD board"
+	depends on DEVEL_BOARD
+	default y
+	help
+	  Choose this option to add
+	  support for Marvell Armada-8K
+	  RD board. Say N unless you really
+	  work on this board
+
+endmenu
diff --git a/board/mvebu/armada8k/Makefile b/board/mvebu/armada8k/Makefile
index 6682204..1c34c15 100644
--- a/board/mvebu/armada8k/Makefile
+++ b/board/mvebu/armada8k/Makefile
@@ -17,5 +17,5 @@
 #
 
 obj-y	+= armada8021-pxp.o
-obj-y	+= db-8210-info.o
-obj-y	+= rd-8210-info.o
+obj-$(CONFIG_APN_806_DB)	+= db-8210-info.o
+obj-$(CONFIG_APN_806_RD)	+= rd-8210-info.o
diff --git a/board/mvebu/armada8k/armada8021-pxp.c b/board/mvebu/armada8k/armada8021-pxp.c
index 8ba2db9..263b802 100644
--- a/board/mvebu/armada8k/armada8021-pxp.c
+++ b/board/mvebu/armada8k/armada8021-pxp.c
@@ -30,14 +30,22 @@
 struct mvebu_board_info *mvebu_board_info_get(enum fdt_compat_id compat_id)
 {
 	switch (compat_id) {
+#ifdef CONFIG_APN_806_DB
 	case COMPAT_MVEBU_APN_806_DB:
 		return &a8021_db_info;
+#endif
+#ifdef CONFIG_APN_806_RD
 	case COMPAT_MVEBU_APN_806_RD:
 		return &a8021_rd_info;
+#endif
 	case COMPAT_MVEBU_APN_806_CUSTOMER:
 	default:
 		error("Missing board information for compatible string = %d\n", compat_id);
+#ifdef CONFIG_APN_806_DB
 		return &a8021_db_info;
+#else
+		return NULL;
+#endif
 	}
 }
 
diff --git a/board/mvebu/armadalp/Kconfig b/board/mvebu/armadalp/Kconfig
new file mode 100644
index 0000000..a48d9d6
--- /dev/null
+++ b/board/mvebu/armadalp/Kconfig
@@ -0,0 +1,34 @@
+menu "Armada LP boards"
+	depends on TARGET_ARMADA_LP
+
+config ARMADA_LP_DB0
+	bool "Support Marvell Armada-8K DB0 board"
+	depends on DEVEL_BOARD
+	default y
+	help
+	  Choose this option to add
+	  support for Marvell Armada-LP
+	  DB0 board. Say N unless you really
+	  work on this board
+
+config ARMADA_LP_DB1
+	bool "Support Marvell Armada-8K DB1 board"
+	depends on DEVEL_BOARD
+	default y
+	help
+	  Choose this option to add
+	  support for Marvell Armada-LP
+	  DB1 board. Say N unless you really
+	  work on this board
+
+config ARMADA_LP_CUSTOMER
+	bool "Support Marvell Armada-8K Customer board"
+	depends on CUSTOMER_BOARD
+	default y
+	help
+	  Choose this option to add
+	  support for Marvell Armada-LP
+	  Customer board. Say N unless you really
+	  work on this board
+
+endmenu
diff --git a/board/mvebu/armadalp/Makefile b/board/mvebu/armadalp/Makefile
index ee1db20..920d906 100644
--- a/board/mvebu/armadalp/Makefile
+++ b/board/mvebu/armadalp/Makefile
@@ -17,6 +17,6 @@
 #
 
 obj-y	+= armadalp.o
-obj-y	+= db0-armadalp-info.o
-obj-y	+= db1-armadalp-info.o
-obj-y	+= customer-board.o
+obj-$(CONFIG_ARMADA_LP_DB0)	+= db0-armadalp-info.o
+obj-$(CONFIG_ARMADA_LP_DB1)	+= db1-armadalp-info.o
+obj-$(CONFIG_ARMADA_LP_CUSTOMER)	+= customer-board.o
diff --git a/board/mvebu/armadalp/armadalp.c b/board/mvebu/armadalp/armadalp.c
index 76dcd90..1d74491 100644
--- a/board/mvebu/armadalp/armadalp.c
+++ b/board/mvebu/armadalp/armadalp.c
@@ -30,14 +30,24 @@
 struct mvebu_board_info *mvebu_board_info_get(enum fdt_compat_id compat_id)
 {
 	switch (compat_id) {
+#ifdef CONFIG_ARMADA_LP_DB0
 	case COMPAT_MVEBU_ARMADA_LP_DB0:
 		return &db0_armadalp_info;
+#endif
+#ifdef CONFIG_ARMADA_LP_DB1
 	case COMPAT_MVEBU_ARMADA_LP_DB1:
 		return &db1_armadalp_info;
+#endif
+#ifdef CONFIG_ARMADA_LP_CUSTOMER
 	case COMPAT_MVEBU_ARMADA_LP_CUSTOMER:
 		return &armadalp_customer_info;
+#endif
 	default:
 		error("Missing board information for compatible string = %d\n", compat_id);
+#ifdef CONFIG_ARMADA_LP_DB0
 		return &db0_armadalp_info;
+#else
+		return NULL;
+#endif
 	}
 }
diff --git a/board/mvebu/common/Kconfig b/board/mvebu/common/Kconfig
deleted file mode 100644
index c7b69d5..0000000
--- a/board/mvebu/common/Kconfig
+++ /dev/null
@@ -1,32 +0,0 @@
-
-menuconfig DEVEL_BOARD
-	bool "Support MVEBU development boards"
-	depends on MVEBU
-	default n
-	---help---
-	  Say Y here to add support for Marvell MVEBU
-	  development boards.
-	  You need this only of you work on a Marvell board.
-	  If not, keep this off to reduce code size
-
-if DEVEL_BOARD
-
-	config ARMADA_38X_DB
-	bool "Support Marvell Armada-38x DB board"
-	default n
-	help
-	  Choose this option to add
-	  support for Marvell Armada-38x
-	  DB board. Say N unless you really
-	  work on this board
-
-	config APN_806_DB
-	bool "Support Marvell APN-806 DB board"
-	default n
-	help
-	  Choose this option to add
-	  support for Marvell APN-806
-	  DB board. Say N unless you really
-	  work on this board
-
-endif #DEVEL_BOARD
diff --git a/board/mvebu/common/Makefile b/board/mvebu/common/Makefile
index 54cb2c7..cc4dd16 100644
--- a/board/mvebu/common/Makefile
+++ b/board/mvebu/common/Makefile
@@ -23,6 +23,5 @@ obj-$(CONFIG_CMD_NET)	+= eth_init.o
 
 ifdef CONFIG_DEVEL_BOARD
 obj-y	+= sar.o
-obj-n	+= config.o
 obj-y	+= devel-board.o
 endif
diff --git a/configs/mvebu_apn806_defconfig b/configs/mvebu_apn806_defconfig
index cd1ed7e..3559a44 100644
--- a/configs/mvebu_apn806_defconfig
+++ b/configs/mvebu_apn806_defconfig
@@ -9,7 +9,7 @@ CONFIG_MVEBU_MPP_BUS=y
 CONFIG_MVEBU_I2C=y
 CONFIG_MVEBU_SPL_SAR_DUMP=y
 +S:CONFIG_DEVEL_BOARD=y
-CONFIG_APN_806_DB=y
+CONFIG_CUSTOMER_BOARD=y
 CONFIG_CMD_BOOTD=y
 CONFIG_CMD_RUN=y
 CONFIG_CMD_EDITENV=y
diff --git a/configs/mvebu_apn806_pd_defconfig b/configs/mvebu_apn806_pd_defconfig
index ac5c447..c33b8eb 100644
--- a/configs/mvebu_apn806_pd_defconfig
+++ b/configs/mvebu_apn806_pd_defconfig
@@ -3,9 +3,7 @@ CONFIG_SYS_EXTRA_OPTIONS="ARM64"
 +S:CONFIG_ARM=y
 +S:CONFIG_TARGET_ARMADA_8K=y
 CONFIG_MVEBU_MPP_BUS=y
-+S:CONFIG_MVEBU_MK6_SUPPORT=y
 +S:CONFIG_DEVEL_BOARD=y
-CONFIG_APN_806_DB=y
 CONFIG_CMD_BOOTD=y
 CONFIG_CMD_RUN=y
 CONFIG_CMD_EDITENV=y
@@ -24,3 +22,4 @@ CONFIG_MVEBU_CCU=y
 CONFIG_MVEBU_RFU=y
 +S:CONFIG_SYS_NS16550=y
 CONFIG_MVEBU_PINCTL=y
++S:CONFIG_MVEBU_MK6_SUPPORT=y
diff --git a/configs/mvebu_armada38x_defconfig b/configs/mvebu_armada38x_defconfig
index 5e13bc3..50a6af1 100644
--- a/configs/mvebu_armada38x_defconfig
+++ b/configs/mvebu_armada38x_defconfig
@@ -7,7 +7,7 @@ CONFIG_MVEBU_PCI=y
 CONFIG_MVEBU_MPP_BUS=y
 +S:CONFIG_MVEBU_I2C=y
 +S:CONFIG_DEVEL_BOARD=y
-+S:CONFIG_ARMADA_38X_DB=y
+CONFIG_CUSTOMER_BOARD=y
 CONFIG_CMD_BDI=y
 CONFIG_CMD_BOOTD=y
 CONFIG_CMD_RUN=y
diff --git a/configs/mvebu_armada7040_pd_defconfig b/configs/mvebu_armada7040_pd_defconfig
index bf1868b..35d156e 100644
--- a/configs/mvebu_armada7040_pd_defconfig
+++ b/configs/mvebu_armada7040_pd_defconfig
@@ -4,7 +4,6 @@ CONFIG_SYS_EXTRA_OPTIONS="ARM64"
 +S:CONFIG_TARGET_ARMADA_8K=y
 CONFIG_MVEBU_MPP_BUS=y
 +S:CONFIG_DEVEL_BOARD=y
-CONFIG_APN_806_DB=y
 CONFIG_CMD_BOOTD=y
 CONFIG_CMD_RUN=y
 CONFIG_CMD_EDITENV=y
diff --git a/configs/mvebu_armada8040_pd_defconfig b/configs/mvebu_armada8040_pd_defconfig
index 72e1cff..cde07c5 100644
--- a/configs/mvebu_armada8040_pd_defconfig
+++ b/configs/mvebu_armada8040_pd_defconfig
@@ -5,7 +5,6 @@ CONFIG_SYS_EXTRA_OPTIONS="ARM64"
 CONFIG_MVEBU_MPP_BUS=y
 +S:CONFIG_MVEBU_MK6_SUPPORT=y
 +S:CONFIG_DEVEL_BOARD=y
-CONFIG_APN_806_DB=y
 CONFIG_CMD_BOOTD=y
 CONFIG_CMD_RUN=y
 CONFIG_CMD_EDITENV=y
diff --git a/configs/mvebu_armadalp_defconfig b/configs/mvebu_armadalp_defconfig
index 094dc0d..4ddb8bf 100644
--- a/configs/mvebu_armadalp_defconfig
+++ b/configs/mvebu_armadalp_defconfig
@@ -2,9 +2,10 @@ CONFIG_SPL=y
 CONFIG_SYS_EXTRA_OPTIONS="ARM64"
 +S:CONFIG_ARM=y
 +S:CONFIG_TARGET_ARMADA_LP=y
-+S:CONFIG_DEVEL_BOARD=y
-+S:CONFIG_MVEBU_UART_ARLP=y
 CONFIG_MVEBU_SPI=y
++S:CONFIG_MVEBU_UART_ARLP=y
++S:CONFIG_DEVEL_BOARD=y
+CONFIG_CUSTOMER_BOARD=y
 CONFIG_CMD_BDI=y
 CONFIG_CMD_SPI=y
 CONFIG_CMD_BOOTD=y
diff --git a/configs/mvebu_armadalp_palladium_defconfig b/configs/mvebu_armadalp_palladium_defconfig
index a8944dd..9c1636d 100644
--- a/configs/mvebu_armadalp_palladium_defconfig
+++ b/configs/mvebu_armadalp_palladium_defconfig
@@ -2,9 +2,9 @@ CONFIG_SPL=y
 CONFIG_SYS_EXTRA_OPTIONS="ARM64"
 +S:CONFIG_ARM=y
 +S:CONFIG_TARGET_ARMADA_LP=y
-+S:CONFIG_DEVEL_BOARD=y
-+S:CONFIG_MVEBU_UART_ARLP=y
 CONFIG_MVEBU_SPI=y
++S:CONFIG_MVEBU_UART_ARLP=y
++S:CONFIG_DEVEL_BOARD=y
 CONFIG_CMD_BDI=y
 CONFIG_CMD_SPI=y
 CONFIG_CMD_BOOTD=y
-- 
1.9.1

